core/src/main/scala/kafka/server/metadata/KRaftMetadataCache.scala

```diff
@@ -174,14 +174,25 @@ class KRaftMetadataCache(val brokerId: Int) extends MetadataCache with Logging w
                                 errorUnavailableEndpoints: Boolean = false,
                                 errorUnavailableListeners: Boolean = false): Seq[MetadataResponseTopic] = {
     val image = _currentImage
-    topics.toSeq.flatMap { topic =>
-      getPartitionMetadata(image, topic, listenerName, errorUnavailableEndpoints, errorUnavailableListeners).map { partitionMetadata =>
+    if (!isInitialized()) {
+      topics.toSeq.map(topic =>
```

Comment:
I think that the BROKER_NOT_AVAILABLE should be a top level error for the metadata API - e.g. for a metadata request, it doesn't seem correct to report it per-partition, when in reality, the broker itself cannot properly serve the request at all (it doesn't even know anything about the topics listed in the request)

Reply:
I agree, however unfortunately the Metadata protocol doesn't implement top level errors ([link](https://github.com/apache/kafka/blob/trunk/clients/src/main/resources/common/message/MetadataResponse.json)) so I tried to go around this. It's also an alternative to return LEADER_NOT_AVAILABLE, although that's rather partition level and not topic level as I understand, but it has the advantage that Java clients don't require any changes to handle it as it's a `RetriableException`.
Would that be better instead?
If you think that is a bit sketchy too I can start a KIP for this to add top level errors to the Metadata API.

Reply:
Oh, I see, didn't know that there is no top level error. I think that would be the ideal solution, but not sure if we need to stick to that.
What if we used UNKNOWN_TOPIC on the topic level instead? That one is already part of the API, and clients are aware of it, too.